Include/objimpl.h

```diff
@@ -309,16 +301,16 @@ extern PyGC_Head *_PyGC_generation0;
  */
 #define _PyObject_GC_UNTRACK(o) do { \
     PyGC_Head *g = _Py_AS_GC(o); \
-    assert(_PyGCHead_REFS(g) != _PyGC_REFS_UNTRACKED); \
-    _PyGCHead_SET_REFS(g, _PyGC_REFS_UNTRACKED); \
-    g->gc.gc_prev->gc.gc_next = g->gc.gc_next; \
-    g->gc.gc_next->gc.gc_prev = g->gc.gc_prev; \
+    PyGC_Head *prev = _PyGCHead_PREV(g); \
+    assert(g->gc.gc_next != NULL); \
+    prev->gc.gc_next = g->gc.gc_next; \
```

Comment:
Hmm... I wonder if the look would less weird if we also had `_PyGCHead_NEXT` and `_PyGCHead_SET_NEXT` macros.